pkg/controller/installation/installation_controller.go

```diff
@@ -80,20 +83,49 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
-	// Watch for changes to secondary resource Pods and requeue the owner Installation
-	err = c.Watch(&source.Kind{Type: &corev1.Pod{}}, &handler.EnqueueRequestForOwner{
-		IsController: true,
-		OwnerType:    &integreatlyv1alpha1.Installation{},
-	})
+	var watchCRDs []runtime.Object
+
+	err, installType := InstallationTypeFactory("managed", []string{"all"})
 	if err != nil {
-		return err
+		return pkgerr.Wrap(err, "could not load managed installation type")
+	}
+
+	configManager, err := config.NewManager(context.Background(), cl, DefaultInstallationConfigMapNamespace, DefaultInstallationConfigMapName, &integreatlyv1alpha1.Installation{})
+	if err != nil {
+		return pkgerr.Wrap(err, "could not load config manager")
 	}
 
+	for _, stage := range installType.GetStages() {
+		for product, _ := range stage.Products {
+			config, err := configManager.ReadProduct(product)
+			if err != nil {
+				continue
+			}
+			watchCRDs = append(watchCRDs, config.GetWatchableCRDs()...)
+
+		}
+	}
+
+	d.AddCRDsTrigger(watchCRDs, func(crd runtime.Object) {
+		logrus.Infof("Detected new CRD to watch: %s", crd.GetObjectKind().GroupVersionKind().String())
+		c.Watch(&source.Kind{Type: crd}, &EnqueueIntegreatlyOwner{})
```

Comment:
If you change your object to implement the `Predicate` interface rather than the `EventHandler` interface, you can make the methods on `EnqueueIntegreatlyOwner` simpler (it's a simpler interface). You'd then change this line to something like this (the predicates come after the EventHandler paremeter, and the EventHandler will only be given events if the predicates return true (so maybe a little more performant (probably negligible if anything)):

```suggestion
		c.Watch(&source.Kind{Type: crd}, &handler.EnqueueRequestForObject{}, &EnqueueIntegreatlyOwner{})
```

See here for where we did it in the unifiedpush-operator previously: https://github.com/aerogear/unifiedpush-operator/commit/19ea331099f81e0823b175a4bdec9272802c93df#diff-6650152505aaf34b2c18559ca4d20db0

If you go that route, `EnqueueIntegreatlyOwner` is probably not the best name for the type then...maybe `IntegreatlyOwnerPredicate`. Ooh, or make it more generic, like `AnnotationsPredicate`, and pass in a map of expected annotations or something. I wonder if the operator-utils project would be a good place for reusable predicates? :slightly_smiling_face:

Reply:
I'm not sure I'm following all of this, let's go on a call and discuss it?

Reply:
discussed on a call, this will not work in this case.